woocommerce/woocommerce
# Enhancement/Add block-shortcode-template usage to Cart and Checkout pages in system report

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #44357 .

This PR adds information about the type of Cart/Checkout in use on the system report tool:
<img width="823" alt="image" src="https://github.com/woocommerce/woocommerce/assets/17236129/34beea34-08be-4f96-a140-378a22ef04b7">

It also adds a warning when both versions are being loaded on the page and adds information about one special case:
By default, on block themes, the templates use the `woocommerce/page-content-wrapper` block to load the respective page content into the page. That way, the page and the template can be edited while being kept in sync. 
This PR also adds a warning if that block is removed (e.g. clearing it out and adding the cart or checkout block), stating that the content is now overridden by the db-saved template.

Apart from that, it introduces a performance improvement preventing multiple repeated queries to the posts tables on the` /wc/v2/system_status` endpoint (and also the system report tool page which relies on it to populate information).

A bug fix is also included, preventing the classic checkout/cart block from being considered as the blocks version (it effectively is the shortcode version).



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. On a fresh woocommerce installation modify the Cart and Checkout pages between the blocks and the shortcode version.
2. Verify that the WooCommerce > Status > WooCommerce Pages section displays information about the currently used version for each page.
<img width="823" alt="image" src="https://github.com/woocommerce/woocommerce/assets/17236129/30e5af5a-6bbe-4651-8fa6-b266f138cd01">

3. Add the Classic Checkout/Cart block to both Checkout and Cart pages.
4.  Verify on WooCommerce > Status > WooCommerce Pages section displays information stating that the shortcode is being used.
5. Add both the blocks version and the shortcode to the pages.
6. Verify on WooCommerce > Status > WooCommerce Pages section a warning is shown stating that both versions are on the same page:
<img width="1136" alt="image" src="https://github.com/woocommerce/woocommerce/assets/17236129/2ff32922-f63f-4773-a7be-aca312884b47">

7. Edit the pages back again and make sure both contain only one version of the Checkout/Cart (either block or shortcode).
7. Switch to a blocks theme, and on the editor for the Page: Checkout and Page:Cart templates delete the `woocommerce/page-content-wrapper` block. Proceed to add the Checkout/Cart block and save the template
8. Verify on WooCommerce > Status > WooCommerce Pages section it displays information stating the the page's content is being overridden by custom template content.
<img width="933" alt="image" src="https://github.com/woocommerce/woocommerce/assets/17236129/5d26b33a-76f3-4fcb-b0e4-d31f6cf6492a">




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Add information about block/shortcode/template usage on Cart and Checkout pages to the WC system report.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
